src/drive/web/modules/viewer/WithDriveLocales.jsx

```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+
+import { I18nContext } from 'cozy-ui/transpiled/react/I18n'
+
+/**
+ * Used to give the translation context of Drive to a cozy-ui component.
+ * @param {object} params - Params
+ * @param {func} params.t - Translate function
+ * @param {node} params.children - React children
+ */
+const WithDriveLocales = ({ t, children }) => {
+  return <I18nContext.Provider value={{ t }}>{children}</I18nContext.Provider>
+}
+
+export default WithDriveLocales
```

Comment:
There is the withLocales HOC available in cozy-ui ? Could it be used here ?

```
import en from 'cozy-drive/locales/en.json'
import en from 'cozy-drive/locales/fr.json'

export const withDriveLocales = withLocales({ en, fr })
```

Reply:
En utilisant withLocales il faut ensuite soit passer `t` à tous les enfants, soit utiliser le HOC sur tous les enfants non ? Ici l'avantage de wrapper dans le provider c'est qu'on peut utiliser simplement `useI18n()` dans les enfants et récupérer les bonnes locales.

Reply:
Non, withLocales met un nouveau contexte avec le bon `t`, donc les enfants aussi récupèrent le bon `t`

Reply:
Raisons pour lesquelles j'ai opté pour un WithDriveLocales et non celui de UI : 
- Dans Drive les locales sont déjà importées, les dictionnaire déjà créés et je me suis dit qu'il fallait mieux les réutiliser plutôt que de nouveau faire un import et une instanciation (via le withLocales de UI). Au final, on fait la même chose que WithLocales, on vient setter le context mais sans devoir réinstancier les dictionnaires

Reply:
A mon avis ca serait plsu simple si le withLocales etait sur les composants du Viewer utilisant directement les locales plutôt que sur tout le viewer, ca éviterait à Drive de devoir faire ça.